lib/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -0,0 +1,124 @@
+
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const validateObjectWithArrayProps = require('../../utils/validateObjectWithArrayProps');
+const isCustomProperty = require('../../utils/isCustomProperty');
+const isStandardSyntaxValue = require('../../utils/isStandardSyntaxValue');
+const isStandardSyntaxProperty = require('../../utils/isStandardSyntaxProperty');
+const isStandardSyntaxDeclaration = require('../../utils/isStandardSyntaxDeclaration');
+const { isRegExp, isString } = require('../../utils/validateTypes');
+
+const ruleName = 'declaration-property-value-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, value) => `Unexpected unknown value "${value}" for property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreProperties: [validateObjectWithArrayProps(isString, isRegExp)],
+					propertiesSyntax: [isPlainObject],
+					typesSyntax: [isPlainObject],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) {
+			return;
+		}
+
+		const ignoreProperties = Array.from(
+			Object.entries((secondaryOptions && secondaryOptions.ignoreProperties) || {}),
+		);
+
+		/** @type {(name: string, propValue: string) => boolean} */
+		const isPropIgnored = (name, value) => {
+			const [, valuePattern] =
+				ignoreProperties.find(([namePattern]) => matchesStringOrRegExp(name, namePattern)) || [];
+
+			return valuePattern && matchesStringOrRegExp(value, valuePattern);
+		};
+
+		const propertiesSyntax = (secondaryOptions && secondaryOptions.propertiesSyntax) || {};
+		const typesSyntax = (secondaryOptions && secondaryOptions.typesSyntax) || {};
+
+		const forkedLexer = fork({
+			properties: propertiesSyntax,
+			types: typesSyntax,
+		}).lexer;
+
+		root.walkDecls((decl) => {
+			const { prop, value } = decl;
+
+			// NOTE: CSSTree's `fork()` doesn't support `-moz-initial`, but it may be possible in the future.
+			// See https://github.com/stylelint/stylelint/pull/6511#issuecomment-1412921062
+			if (/^-moz-initial$/i.test(value)) return;
+
+			if (!isStandardSyntaxDeclaration(decl)) return;
+
+			if (!isStandardSyntaxProperty(prop)) return;
+
+			if (!isStandardSyntaxValue(value)) return;
+
+			if (isCustomProperty(prop)) return;
+
+			if (isPropIgnored(prop, value)) return;
+
+			try {
+				parse(value, {
+					context: 'value',
+				});
+			} catch (e) {
+				return;
```

Comment:
[suggest] What about handling a parse error like this?

```suggestion
				result.warn(`Cannot parse property value "${value}"`, {
					node: decl,
					stylelintType: 'parseError',
				});

				return;
```

If the following code would be given:

```css
a { color: &a; }
```

the rule could report:

```
Cannot parse property value "&a"
```